lib/IRGen/GenStruct.cpp

```diff
@@ -61,6 +63,31 @@ static StructTypeInfoKind getStructTypeInfoKind(const TypeInfo &type) {
   return (StructTypeInfoKind) type.getSubclassKind();
 }
 
+clang::CXXDestructorDecl *swift::irgen::getCXXDestructor(SILType type) {
+  clang::CXXDestructorDecl *destructor = nullptr;
+  if (!type.getStructOrBoundGenericStruct() ||
+      !type.getStructOrBoundGenericStruct()->getClangDecl())
+    return nullptr;
+  const clang::CXXRecordDecl *cxxRecordDecl = dyn_cast<clang::CXXRecordDecl>(type.getStructOrBoundGenericStruct()->getClangDecl());
+  if (!cxxRecordDecl)
+    return nullptr;
+  for (auto member : cxxRecordDecl->methods()) {
+    if (auto dest = dyn_cast<clang::CXXDestructorDecl>(member)) {
+      destructor = dest;
+      break;
+    }
+  }
+  return destructor;
+}
+
+llvm::Constant *
+swift::irgen::getCXXDestructorCallee(IRGenModule &IGM,
+                                  clang::CXXDestructorDecl *destructor) {
```

Comment:
Given that this is used only once (I think?), I'm not sure it's worth factoring this out into a separate function.

It certainly doesn't look like this needs to be a member function (I think it doesn't access any members?), so if you want to keep it, make it a static non-member function instead.